machine.go

```diff
@@ -667,21 +685,39 @@ func (m *Machine) startup(command string, extracontent [][2]string) (int, error)
 		return -1, err
 	}
 
-	/* Amd64 dynamic linker */
-	err = w.CopyFile("/lib64/ld-linux-x86-64.so.2")
+	var dynamicLinker string
+	if m.arch == Arm64 {
+		// arm64 dynamic linker is in /lib:
+		// https://sourceware.org/bugzilla/show_bug.cgi?id=25129
+		dynamicLinker = "/lib/ld-linux-aarch64.so.1"
+	} else {
+		dynamicLinker = "/lib64/ld-linux-x86-64.so.2"
+	}
+
+	/* dynamic linker */
+	err = w.CopyFile(dynamicLinker)
 	if err != nil {
 		return -1, err
 	}
 
 	/* C libraries */
-	libraryDir, err := realDir("/lib64/ld-linux-x86-64.so.2")
+	libraryDir, err := realDir(dynamicLinker)
 	if err != nil {
 		return -1, err
 	}
+
 	err = w.CopyFile(libraryDir + "/libc.so.6")
+	if err != nil && errors.Is(err, os.ErrNotExist) && m.arch == Arm64 && strings.HasSuffix(libraryDir, "/lib") {
```

Comment:
would it be worth making the check for arch more clear ?
or even, do we need the arch check here ?

Reply:
This needs a more generic fix to get this to work on fedora.

The overall assumption this code makes is that the main library directory *matches* the real location of the dynamic linker.

This is true  for debian where the canonical linker locations are  e.g. "/lib/ld-linux-aarch64.so.1" or "/lib64/ld-linux-x86-64.so.2" are symlinks to "
aarch64-linux-gnu/ld-linux-aarch64.so.1" and "/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2" respectively.. 

It also works for arch as it only seems to use `/lib64` on amd64 where it's just a symlink to `/usr/lib` anyway.

So fedora is the odd one out here in that it uses `/usr/lib64`  on arm64 *as well* and having it's dynamic linkers  directly in their canonical location.. Which causes the failure on arm64 as the linker in `lib` but everything else in `lib64`

TL;DR to cover fedora (and the likes) properly the way we determine libraryDir should change/be expanded, and probably broken out to an helper, rather then patching it up after the fact